lib/core/src/Cardano/Wallet/DB/Layer.hs

```diff
@@ -577,40 +557,31 @@ newDBLayerWith _cacheBehavior _tr ti SqliteContext{runQuery} = do
                                       Wallets
         -----------------------------------------------------------------------}
 
-        { initializeWallet = \wid cp meta txs gp -> ExceptT $ do
-            res <- handleConstraint (ErrWalletAlreadyExists wid) $
-                insert_ (mkWalletEntity wid meta gp)
-            when (isRight res) $ do
-                insertCheckpointGenesis wid cp
-                let Identity
-                        ( txMetas
-                        , txIns
-                        , txCollateralIns
-                        , txOuts
-                        , txOutTokens
-                        , txCollateralOuts
-                        , txCollateralOutTokens
-                        , txWithdrawals
-                        ) = Identity $ mkTxHistory wid txs
-                putTxs
-                    txMetas
-                    txIns
-                    txCollateralIns
-                    txOuts
-                    txOutTokens
-                    txCollateralOuts
-                    txCollateralOutTokens
-                    txWithdrawals
-            pure res
-
-        , removeWallet = \wid -> ExceptT $ do
-            selectWallet wid >>= \case
-                Nothing -> pure $ Left $ ErrNoSuchWallet wid
-                Just _  -> Right <$> do
-                    deleteWhere [WalId ==. wid]
-                    deleteLooseTransactions
-                    deleteCheckpoints wid
-
+        { initializeWallet = \wid cp meta txs gp -> do
+            ExceptT $ do
+                res <- handleConstraint (ErrWalletAlreadyExists wid) $
+                    insert_ (mkWalletEntity wid meta gp)
+                when (isRight res) $ do
+                    insertCheckpointGenesis wid cp
+                    void $ modifyDBMaybe transactionsDBVar $ \(_txsOld, _ws) ->
+                        let delta = Just $ ExpandTxWalletsHistory wid txs
+                        in  (delta, Right ())
+                pure res
+        , removeWallet = \wid -> do
+            ExceptT $ do
+                selectWallet wid >>= \case
+                    Nothing -> pure $ Left $ ErrNoSuchWallet wid
+                    Just _  -> Right <$> do
+                        deleteWhere [WalId ==. wid]
+                        deleteCheckpoints wid
+            ExceptT $ modifyDBMaybe transactionsDBVar $ \_ ->
+                        let
+                            delta = Just $ RemoveWallet wid
+                        in  (delta, Right ())
+            ExceptT $ modifyDBMaybe transactionsDBVar $ \_ ->
```

Comment:
I suggest to join these two `modifyDBMaybe` by using a list of deltas as the delta type for the DBVar — However, I recommend to do that in a future PR; this is completely fine for now.

Reply:
future PR approved!